crates/noirc_evaluator/src/ssa_refactor/ir/value.rs

```diff
@@ -40,6 +40,9 @@ pub(crate) enum Value {
     /// Represents a constant array value
     Array { array: im::Vector<ValueId>, element_type: Rc<CompositeType> },
 
+    /// Represents a compile-time array
+    Slice { array: im::Vector<ValueId>, element_type: Rc<CompositeType> },
```

Comment:
The only difference between this value and array is that the type of Slice is a Reference

Reply:
Only when we use `get_type` really. Technically the same Value could be reused I believe. I will try and see if it causes any issues

Reply:
I removed the `Slice` value